src/main/java/ludum/dare/screens/FightScreen.java

```diff
@@ -0,0 +1,262 @@
+
+//        for (GameObject gameObj : gameObjects) {
+//            if (gameObj instanceof SpawnGameObject) {
+//                SpawnGameObject spawn = (SpawnGameObject) gameObj;
+//                for (Player player : Players.list()) {
+//                    player.setPosition(spawn.pos.x, spawn.pos.y);
+//                    player.addToWorld(world);
+//                }
+//            }
+//        }
+
+        for (Player player : Players.list()) {
+            player.activateControls();
+            player.addToWorld(world);
+            // TODO handle spawn points.
+            player.setPosition(0, 0);
+            gameObjects.add(player);
+        }
+
+        if (level.debugSpawn != null) {
+            JumperBody playerBody = new JumperBody();
+            playerBody.props = level.debugSpawn.props;
+            playerBody.jumperProps = level.debugSpawn.jumpProps;
+
+            playerBody.bodyType = BodyType.DYNAMIC;
+            playerBody.aabb = new BitRectangle(level.debugSpawn.rect.xy.x, level.debugSpawn.rect.xy.y, 16, 32);
+            playerBody.renderStateWatcher = new JumperRenderStateWatcher();
+            playerBody.controller = new PlayerInputController(GDXControls.defaultMapping);
+
+            playerBody.addContactListener(new ContactAdapter() {
+                @Override
+                public void contactStarted(BitBody other) {
+                    if (other.userObject instanceof SecretObject) {
+                        playerBody.props.gravityModifier = -1;
+                    }
+                }
+
+                @Override
+                public void contactEnded(BitBody other) {
+                    if (other.userObject instanceof SecretObject) {
+                        playerBody.props.gravityModifier = 1;
+                    }
+                }
+            });
+
+            world.addBody(playerBody);
+        }
+    }
+
+    private Collection<BitBody> buildBodies(Collection<LevelObject> otherObjects) {
+        try {
+            ArrayList<BitBody> bodies = new ArrayList<>();
+            for (LevelObject levelObject : otherObjects) {
+                if (builderMap.containsKey(levelObject.getClass())) {
+                    GameObject newObject;
+                    newObject = ((GameObject) builderMap.get(levelObject.getClass()).newInstance());
+                    bodies.addAll(newObject.build(levelObject));
+                    gameObjects.add(newObject);
+                } else {
+                    bodies.add(levelObject.buildBody());
+                }
+            }
+            return bodies;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public void resize(int i, int i1) {
+
+    }
+
+    @Override
+    public void pause() {
+
+    }
+
+    @Override
+    public void resume() {
+
+    }
+
+    @Override
+    public void hide() {
+
+    }
+
+    @Override
+    public void dispose() {
+
+    }
+}
```

Comment:
We're going to need t0 find a way to merge common logic in `RaceScreen` and `FightScreen`, otherwise it's going to be hell for us (and you guys eventually) to add new features to both in an ongoing manner. `AbstractLevelScreen` class?